syntax = "proto3";

service TodoService {
  rpc AddTask (AddTaskRequest) returns (AddTaskResponse);
  rpc GetTasks (GetTasksRequest) returns (GetTasksResponse);
  rpc CompleteTask (CompleteTaskRequest) returns (CompleteTaskResponse);
}

message AddTaskRequest {
  string title = 1;
}

message AddTaskResponse {
  bool success = 1;
}

message GetTasksRequest {}

message GetTasksResponse {
  repeated Task tasks = 1;
}

message CompleteTaskRequest {
  int32 id = 1;
}

message CompleteTaskResponse {
  bool success = 1;
}

message Task {
  int32 id = 1;
  string title = 2;
  bool completed = 3;
}
